File: src/console/log/output/trait/AuthorizedLoggerBase.php
<?php

namespace pocketcloud\cloud\console\log\output\trait;

use pocketcloud\cloud\console\log\logger\ILogger;

trait AuthorizedLoggerBase {

    private array $authorizedLoggers = [];

    public function addAuthorizedLogger(ILogger $logger): void {
        $this->authorizedLoggers[spl_object_id($logger)] = true;
    }

    public function removeAuthorizedLogger(ILogger $logger): void {
        unset($this->authorizedLoggers[spl_object_id($logger)]);
    }

    public function isAuthorizedLogger(ILogger $logger): bool {
        return isset($this->authorizedLoggers[spl_object_id($logger)]);
    }
}

File: src/main/java/de/pocketcloud/cloud/console/output/util/AuthorizedLoggerBase.java
package de.pocketcloud.cloud.console.output.util;

import de.pocketcloud.cloud.console.log.ILogger;

import java.util.ArrayList;
import java.util.List;

public abstract class AuthorizedLoggerBase {

    protected final List<ILogger> authorizedLoggers = new ArrayList<>();

    public void add(ILogger logger) {
        authorizedLoggers.add(logger);
    }

    public void remove(ILogger logger) {
        authorizedLoggers.remove(logger);
    }

    public boolean isAuthorized(ILogger logger) {
        return authorizedLoggers.contains(logger);
    }
}